src/internal/connector/graph/service_helper.go

```diff
@@ -68,12 +62,39 @@ type LoggingMiddleware struct{}
 
 // Intercept implements the RequestInterceptor interface and decodes the parameters name
 func (handler *LoggingMiddleware) Intercept(
-	pipeline khttp.Pipeline, middlewareIndex int, req *nethttp.Request,
+	pipeline khttp.Pipeline,
+	middlewareIndex int,
+	req *nethttp.Request,
 ) (*nethttp.Response, error) {
-	requestDump, _ := httputil.DumpRequest(req, true)
-	logger.Ctx(context.TODO()).Infof("REQUEST: %s", string(requestDump))
+	var (
+		ctx       = req.Context()
+		resp, err = pipeline.Next(req, middlewareIndex)
+	)
+
+	if resp == nil {
+		return resp, err
+	}
+
+	if (resp.StatusCode / 100) == 2 {
```

Comment:
I feel like just having non 2xx along might be misleading/hard when we try to walk trough and figure out what might be going on. Should we provide an additional flag to also log 2xx requests. Something like `-v` `-vv` pattern.

Reply:
I don't think that's necessary at the moment.  In past experience its been largely unhelpful to retain happy calls.  On one hand, those successes are often captured by other info/debug logs, and on the other it's rare that a successful api call provides detail useful for investigation.

What might appeal to you, however, is that we will be adding some telemetry here to track overall call health (success rates, time, etc).  This is likely to give us much of the same insight.